### PR TITLE
espressif: clean up unused code

### PR DESCRIPTION
**Pull request by sylvioalves**
Remove all entries that as not being used.
This also update hal to re-enable warning flags
as such as -Wno-unused-variable.

### CODE REVIEW

File: drivers/regulator/regulator_axp192.c [hunk 1]
```diff
@@ -64,7 +64,7 @@ static const struct linear_range dcdc1_ranges[] = {
 	LINEAR_RANGE_INIT(700000U, 25000U, 0x00U, 0x7FU),
 };
 
-static const struct regulator_axp192_desc dcdc1_desc = {
+__maybe_unused static const struct regulator_axp192_desc dcdc1_desc = {
```
Comment (sylvioalves) [inline drivers/regulator/regulator_axp192.c:67]: This regulator is only used in M5stick/M5Stack boards. Not all of this descriptors are used, causing compliance check errors. So make it `__maybe_unused` to overcome it.

Comment (uLipe) [reply drivers/regulator/regulator_axp192.c:67]: I'm wondering if there is a way to address these regulator stuff via device tree.

To be clear, I'm not against of the maybe_unused thing.

**Review by uLipe — DISMISSED** [review]

File: drivers/clock_control/clock_control_esp32.c [hunk 2]
```diff
@@ -687,14 +686,13 @@ static int esp32_cpu_clock_configure(const struct esp32_cpu_clock_config *cpu_cf
 	esp_cpu_set_cycle_count((uint64_t)esp_cpu_get_cycle_count() * rtc_clk_cfg.cpu_freq_mhz /
 				old_config.freq_mhz);
 
+#if !defined(ESP_CONSOLE_UART_NONE)
```
Comment (raffarost) [inline drivers/clock_control/clock_control_esp32.c:689]: just a note to check the CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG case (if necessary)
as the routine `esp_rom_uart_set_clock_baudrate` mentions UART I couldn't be sure about the USB clock case

File: drivers/spi/spi_esp32_spim.c [hunk 3]
```diff
@@ -331,6 +330,10 @@ static int IRAM_ATTR spi_esp32_configure(const struct device *dev,
 	hal_dev->cs_pin_id = ctx->config->slave;
 	int ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
 
+	if (ret) {
+		return ret;
```
Comment (raffarost) [inline drivers/spi/spi_esp32_spim.c:334]: an error message could be nice here

**Review by raffarost — DISMISSED** [review]

File: drivers/clock_control/clock_control_esp32.c (hunk 2 above)
Comment (sylvioalves) [reply drivers/clock_control/clock_control_esp32.c:689]: This needs to be handled in another PR.

**Review by raffarost — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Pull request merged**